### PR TITLE
Add pre-commit hook.

### PR DESCRIPTION
**Pull request by buh**
#skip_changelog

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -7,12 +7,18 @@
...
-set -eu
 ln -s ../../hooks/pre-commit.sh .git/hooks/pre-commit
 chmod +x .git/hooks/pre-commit
-echo "Symlink created successfully"
+echo "Symlink for pre-commit created successfully"
+
+# Add symlink for git-format-staged
+ln -s ../../hooks/git-format-staged .git/hooks/git-format-staged
+chmod +x .git/hooks/git-format-staged
+echo "Symlink for git-format-staged created successfully"
```
Comment (b-onc) [inline bootstrap.sh:21]: This is not a git hook, so this does not need a symlink

File: hooks/pre-commit.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+./git-format-staged --formatter "swiftformat --config .swiftformat stdin" "../../StreamChatClient_v3/*.swift"
```
Comment (b-onc) [inline hooks/pre-commit.sh:2]: Instead of `./git-format-staged`, you should use `../../hooks/git-format-staged` (after you remove unnecessary symlink)

**Comment by VojtaStavik** [comment]
I checked out the branch, I ran `bootstrap.sh` and tried to commit something. Got this error:

Screen Shot 2020-06-25 at 17 21 32

File: hooks/pre-commit.sh [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
-./hooks/git-format-staged --formatter "swiftformat --config .swiftformat stdin" "StreamChatClient_v3/**/*.swift"
-./hooks/git-format-staged --formatter "swiftformat --config .swiftformat stdin" "TestResources_v3/**/*.swift"
+./hooks/git-format-staged --formatter 'swiftformat --config .swiftformat stdin' 'StreamChatClient_v3/*.swift'
+./hooks/git-format-staged --formatter 'swiftformat --config .swiftformat stdin' 'TestResources_v3/*.swift'
```
Comment (buh) [inline hooks/pre-commit.sh:3]: @b-onc will it work for subfolders?

Comment (b-onc) [reply hooks/pre-commit.sh:3]: Yes, tested and working. Please test yourself too, so we're sure it's working!

**Pull request merged**